spec/glossary.adoc

```diff
@@ -2,12 +2,23 @@
+[[finding]]Finding:: Generic concept that includes all things reported by <<sonar_analyzer,Sonar analyzers>>: issues, security hotspots, taint vulnerabilities, ...
+[[finding_context]]Finding Context:: When reporting a finding, <<sonar_analyzer,Sonar analyzers>> can optionally provide a context key. This context key will be used to adapt the <<rule_details>> to give more specific remediation guidelines. For example, we can have framework specific context (spring vs struts).
 [[project_key]]Project Key:: Unique identifier of a <<sonar_project,Sonar project>> in a given SonarQube or SonarCloud instance. Project key can be changed on server side (the server use uuid internally).
+[[raw_rule_details]]Raw Rule Details:: Original <<rule_details>> as provided by <<sonar_analyzer>>.
+[[rule]]Rule:: A coding standard or practice which should be followed. <<sonar_analyzer,Sonar analyzers>> will analyze the user's code and report <<finding,findings>> for each rule. Each <<sonar_analyzer,Sonar analyzer>> will contribute a set of rules, identified by a unique <<rule_key,rule key>>. Rules have attached <<rule_details,details>> (name, type, tags, description, default activation...). Rules can also have parameters, with default values. On SonarQube or SonarCloud, users can create quality profiles, that are logical rule sets in which they can decide to activate some rules, and override certain rules metadata (severity, parameters, ...). <<sonar_project,Sonar projects>> are associated with a quality profile. When analyzing the code of a project, <<sensor,sensors>> are only interested in enabled rules and the effective parameter values. This is what we call <<active_rule,active rules>>.
+[[rule_details]]Rule Details:: All the attributes attached to a rule, like <<rule_key,key>>, name, severity, description, ... Some attributes can be overridden at different levels. For example severity and parameter values can be changed in the quality profile. The rule description can also be sensitive to the finding <<finding_context,context>>. See <<raw_rule_details>> and <<effective_rule_details>>.
+[[rule_key]]Rule Key:: Unique identifier of a <<rule,rule>>. The format of a rule key is always [<<rule_repo,rule_repo_key>>]:[rule_id]. For example: `java:S1423`.
+[[rule_repo]]Rule Repository:: A rule repository is a logical grouping of rules. This is not a concept exposed in SonarLint. It is mostly used to differentiate rules of the same language that are assessed by different <<sonar_analyzer, analyzers>> or <<sensor,sensors>>.
+[[sensor]]Sensor:: A Sensor is the Java interface implemented by <<sonar_analyzer,Sonar analyzers>> to analyze source files and report <<finding,findings>>.
+[[sonar_analyzer]]Sonar Analyzer:: A component responsible to analyze user's code, for one or multiple (usually related) languages.
```

Comment:
Not sure I understand the "(usually related)" part

Reply:
removed